File: R/algorithmic_toolbox/10_changing_money.R
#!/usr/bin/env Rscript

coins <- c(10, 5, 1)

CalcMinimumNumberOfChanges <- function(amount) {
    numberOfChanges <- 0
    for (coin in coins) {
        numberOfChanges <- numberOfChanges + amount %/% coin
        amount <- amount %% coin
    }
    numberOfChanges
}

Main <- function() {
    conn <- file("stdin", "r")
    amount <- as.integer(readLines(conn, n=1))
    close(conn)
    result <- CalcMinimumNumberOfChanges(amount)
    cat(result, fill = TRUE)
}

if (!interactive()) {
    Main()
}

File: Python/algorithmic_toolbox/10_changing_money.py
import sys

COINS = (10, 5, 1)


def calc_minimum_number_of_changes(amount):
    number_of_changes = 0
    for coin in COINS:
        number_of_changes += (amount // coin)
        amount = amount % coin
    return number_of_changes


def main():
    data = sys.stdin.read().splitlines()
    amount = int(data[0])
    result = calc_minimum_number_of_changes(amount)
    print(result)


if __name__ == "__main__":
    main()
